OpenRA.Mods.Common/Traits/ResupplyManager.cs

```diff
@@ -49,7 +52,7 @@ public ResupplyManager(Actor self, ResupplyManagerInfo info)
 		void INotifyCreated.Created(Actor self)
 		{
 			health = self.TraitOrDefault<Health>();
-			ammoPools = self.TraitsImplementing<AmmoPool>().ToArray();
+			ammoPools = self.TraitsImplementing<AmmoPool>().Where(x => !x.AutoReloads).ToArray();
```

Comment:
Why not leaving it as IEnumerable instead of converting it to an array?

Reply:
Because `Array` is the simplest implementation and `IEnumerable` runs the risk of reenumerating the thing, making this a failed attempt at caching.